src/main/java/com/ericsson/ei/waitlist/WaitListWorker.java

```diff
@@ -66,33 +66,43 @@ public TaskScheduler taskScheduler() {
-        List<String> documents = waitListStorageHandler.getWaitList();
-        for (String document : documents) {
-            JSONObject eventObject = new JSONObject(document);
-            if (eventToObjectMapHandler.isEventInEventObjectMap(eventObject.getString("_id"))) {
-                waitListStorageHandler.dropDocumentFromWaitList(document);
-            } else {
-                String event = eventObject.getString("Event");
-                rulesObject = rulesHandler.getRulesForEvent(event);
-                String idRule = rulesObject.getIdentifyRules();
-
-                if (idRule != null && !idRule.isEmpty()) {
-                    JsonNode ids = jmesPathInterface.runRuleOnEvent(idRule, event);
-                    if (ids.isArray()) {
-                        LOGGER.debug("[EIFFEL EVENT RESENT] id:" + eventObject.getString("_id") + " time:"
-                                + eventObject.getString("Time"));
-                        for (final JsonNode idJsonObj : ids) {
-                            Collection<String> objects = matchIdRulesHandler.fetchObjectsById(rulesObject,
-                                    idJsonObj.textValue());
-                            if (!objects.isEmpty()) {
-                                rmqHandler.publishObjectToWaitlistQueue(event);
-                            }
-                        }
-                    }
-                }
-            }
+    public void run(){
+    	RulesObject rulesObject;
+    	List<String> documents = waitListStorageHandler.getWaitList();
+    	for (String document : documents) {
+    		try {
+    			ObjectMapper objectMapper = new ObjectMapper();
+    			JsonNode eventJson = objectMapper.readTree(document);
+    			String id = eventJson.get("_id").asText();
+    			if (eventToObjectMapHandler.isEventInEventObjectMap(id)) {
+    				waitListStorageHandler.dropDocumentFromWaitList(document);
+    			} else {
+    				JsonNode event = eventJson.get("Event");
+    				String eventStr = event.asText();
+    				rulesObject = rulesHandler.getRulesForEvent(eventStr);
+    				String idRule = rulesObject.getIdentifyRules();
+
+    				if (idRule != null && !idRule.isEmpty()) {
+    					JsonNode ids = jmesPathInterface.runRuleOnEvent(idRule, eventStr);
+    					if (ids.isArray()) {
+    						JsonNode idNode = eventJson.get("_id");
+    						JsonNode timeNode = eventJson.get("Time");
+    						LOGGER.debug("[EIFFEL EVENT RESENT] id:" + idNode.textValue() + " time:"
+    								+ timeNode);
+    						for (final JsonNode idJsonObj : ids) {
+    							Collection<String> objects = matchIdRulesHandler.fetchObjectsById(rulesObject,
+    									idJsonObj.textValue());
+    							if (!objects.isEmpty()) {
+    								rmqHandler.publishObjectToWaitlistQueue(eventStr);
+    							}
+    						}
+    					}
+    				}
+    			}
+    		} catch (Exception e) {
+    			LOGGER.error("Exception occured while trying to resend event: " + document);
+    			e.printStackTrace();
```

Comment:
printStackTrace can be added to the error message, by adding "e" to the error message, I think.